### DISCUSSION TITLE
Question about AVC sequence header

### DISCUSSION THREAD

**Discussion by levs42** [HaishinKit/HaishinKit.swift#1451] [Q&A]
There's an open source RTMP server that I use for quick testing in the local network: [NMS](https://github.com/illuspas/Node-Media-Server). It parses AVC sequence header when video data arrives [here](https://github.com/illuspas/Node-Media-Server/blob/abcacc3b9274cfa6df8aab874df2c255379f710c/src/node_rtmp_session.js#L789) and extracts profile, level, width and height [here](https://github.com/illuspas/Node-Media-Server/blob/abcacc3b9274cfa6df8aab874df2c255379f710c/src/node_core_av.js#L269) from the AVC header. When I'm trying to publish using HK, that method fails to extract the data from the header. Looks like the muxer doesn't [include](https://github.com/shogo4405/HaishinKit.swift/blob/3e1604ee6bcabe1dfdeb547610cc50137afca662/Sources/RTMP/RTMPMuxer.swift#L42) much into the header. `AVCDecoderConfigurationRecord` defines multiple different fields to be included. Would it be beneficial to do so?

**Comment by shogo4405** [comment]
https://github.com/shogo4405/HaishinKit.swift/blob/3e1604ee6bcabe1dfdeb547610cc50137afca662/Sources/RTMP/RTMPMuxer.swift#L43
I understand that the configurationBox value is the value of AVCDecoderConfigurationRecord and is being sent.

This value needs to be used as output by VideoToolbox,
and changing it is not recommended because it might prevent playback on the viewer side.

I think the reason why NMS can't parse it lies elsewhere.

**Reply by levs42** [reply]
Thank you, I'll look into it. So far it looks a bit strange, "This key is used by sample description bridges to hold sample description extension atoms that they do not recognize."
